### PR TITLE
WT-6857 Define history store cursor

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/cursor/cur_hs.c [hunk 1]
```diff
@@ -141,3 +141,100 @@ __wt_hs_cursor_search_near(WT_SESSION_IMPL *session, WT_CURSOR *cursor, int *exa
...
+
+/*
+ * __curhs_close --
+ *     WT_CURSOR->close method for the hs cursor type.
+ */
+static int
+__curhs_close(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_HS *hsc;
```
Comment (hbokhari) [inline src/cursor/cur_hs.c:153]: I feel like name `hsc` is not very intuitive. Maybe change it to `hs_cursor`?

Same comment for `__curhs_reset` and `__wt_curhs_open`

File: src/cursor/cur_hs.c [hunk 2]
```diff
@@ -141,3 +141,100 @@ __wt_hs_cursor_search_near(WT_SESSION_IMPL *session, WT_CURSOR *cursor, int *exa
...
+err:
+    API_END_RET(session, ret);
+}
+
+/*
+ * __wt_curhs_open --
+ *     Initialize a history store cursor.
+ */
+int
+__wt_curhs_open(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
```
Comment (hbokhari) [inline src/cursor/cur_hs.c:198]: What is happening with `cursorp` when the function call is successful?

File: src/cursor/cur_hs.c (hunk 1 above)
Comment (quchenhao) [reply src/cursor/cur_hs.c:153]: I'm following the naming convention of other cursor types. I agree hsc is a bit short. I'll change that. I think for the function naming we need to be consistent for now.

File: src/cursor/cur_hs.c (hunk 2 above)
Comment (quchenhao) [reply src/cursor/cur_hs.c:198]: Yeah, I missed that. Good catch.

File: src/cursor/cur_hs.c (hunk 1 above)
Comment (hbokhari) [reply src/cursor/cur_hs.c:153]: I meant to say that the same variable name has been used in `__curhs_reset` and `__wt_curhs_open`. I have no objection on the function naming - sorry for the confusion.

File: src/cursor/cur_hs.c (hunk 2 above)
Comment (quchenhao) [reply src/cursor/cur_hs.c]: Fixed.

**Review by hbokhari — APPROVED** [review]
LGTM

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**